### DISCUSSION TITLE
[fixed] id column is overwritten automatically

### DISCUSSION THREAD

**Discussion by MedLabs** [pocketbase/pocketbase#607] [Q&A] [answered]
I wanted to generate my custom ids, but each time I POST data to pb the id is overwritten. Even if my id is a string with length 15.
created and updated are overwritten too.

Should I just create a _id column and createdAt updatedAt and ignore the default ones ?

[EDIT] : it was my fault, I'm using old version of pocketbase executable (custom IDs was introduced in 0.4.x)

**Comment by ganigeorgiev** [comment]
I'm not able to reproduce it. Just tested creating a new record with custom ID and it works as expected for me.

**Please note that the ID can be set only on create and once set it cannot change** (unless you modify manually the record in the database).

`created` and `updated` are metafields that are managed internally. They cannot be modified via the REST API (or SDK). You can define your own timestamp fields if you need custom values.

**Reply by ganigeorgiev** [reply]
We also have integration tests for the id submission (https://github.com/pocketbase/pocketbase/blob/master/apis/record_test.go#L754-L807).

@MedLabs Could you provide a code sample how you are sending the request?

**Comment by MedLabs** [comment]
![Screenshot from 2022-09-22 21-01-12](https://user-images.githubusercontent.com/6123009/191840439-6f8f27ff-0fa3-49c1-9c9b-279d338f817f.png)

**Reply by ganigeorgiev** [reply]
I'm not able to reproduce it. What version of PocketBase are you using?

**Comment by MedLabs** [comment]
0.7.0

**Comment by MedLabs** [comment, marked as answer]
You mean the js-sdk ? 
Wait I just realized that I'm using the same ./pocketbase_0.3.2_linux_amd64/pocketbase :scream:

**Reply by ganigeorgiev** [reply]
You have to upgrade the executable if you need support for custom insertion ids because they were added in v0.4.x+ (**make sure to have a backup of your `pb_data` before running the new executable**).

**Reply by MedLabs** [reply]
exactly I've downloaded the latest executable.
Thanks for this great piece of art, and sorry for this fake issue
